src/twisted/internet/_resolver.py

```diff
@@ -247,26 +246,25 @@ def resolveHostName(
 
         resolution = HostResolution(hostName)
         resolutionReceiver.resolutionBegan(resolution)
-        onAddress = self._simpleResolver.getHostByName(hostName)
-
-        def addressReceived(address: str) -> None:
-            resolutionReceiver.addressResolved(IPv4Address("TCP", address, portNumber))
-
-        def errorReceived(error: Failure) -> None:
-            if not error.check(DNSLookupError):
-                self._log.failure(
+        (
```

Comment:
maybe it's best to have this as inline callbacks?
I am not a fan of LISP and big nested parenthesis and chained functions.

Reply:
There's a teetering jenga tower of type inference happening here and I'd prefer to keep it as-is for the moment rather than `async def`-ifying it just for aesthetics.  Happy to do a follow-up PR about this though.